File: server/src/main/java/rs/ac/singidunum/novisad/server/model/nastavnik/Zvanje.java
package rs.ac.singidunum.novisad.server.model.nastavnik;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Zvanje {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDateTime datumIzbora;
    private LocalDateTime datumPrestanka;

    @ManyToOne
    private Nastavnik nastavnik;
    @OneToOne
    private TipZvanja tipZvanja;
    @OneToOne
    private NaucnaOblast naucnaOblast;
}

File: client/src/app/model/zvanje.ts
import {Nastavnik} from "./nastavnik";
import {TipZvanja} from "./tipZvanja";
import {NaucnaOblast} from "./naucnaOblast";

export interface Zvanje {
  id: number |null,
  datumIzbora: Date|null,
  datumPrestanka: Date |null,
  nastavnik: Nastavnik | null,
  tipZvanja: TipZvanja,
  naucnaOblast: NaucnaOblast|null
}
